### PR TITLE
Logicmonitorexporter implementation

### PR DESCRIPTION
**Pull request by khyatigandhi6**
**Description:**
Adding new component logicmonitorexporter for exporting traces and logs to Logicmonitor.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13727

**Testing:** Added unit test cases.

**Documentation:** Details are mentioned in README.md file.

### CODE REVIEW

**Comment by lalit-fgr** [comment]
@khyatigandhi6 can you please resolve merge conflicts

**Comment by lalit-fgr** [comment]
@Aneurysm9 Can you please review this

**Comment by TylerHelmuth** [comment]
@dmitryax are you the rotating sponsor for this component?

**Comment by khyatigandhi6** [comment]
Hi @Aneurysm9 ,
Can you please assign this PR to the sponsor of this component for reviewing ?

**Comment by Aneurysm9** [comment]
Who is the sponsor of this component?  I still see #13727 tagged as needing a sponsor.

**Comment by jpkrohling** [comment]
@bogdandrutu, it looks like you are the assigned sponsor for this one.

**Comment by lalit-fgr** [comment]
@khyatigandhi6 can you please resolve the merge conflicts
@bogdandrutu a gentle reminder, if you can please this PR.

File: .github/ALLOWLIST [hunk 1]
```diff
@@ -17,6 +17,7 @@ cmd/configschema
 examples/demo/client
 examples/demo/server
 exporter/influxdbexporter
+exporter/logicmonitorexporter
```
Comment (mx-psi) [inline .github/ALLOWLIST:20]: New components must not be added to the allowlist: they must have an owner

**Review by mx-psi — CHANGES_REQUESTED** [review]

File: .github/ALLOWLIST (hunk 1 above)
Comment (khyatigandhi6) [reply .github/ALLOWLIST]: Removed the entry from allowlist. Thanks!

**Comment by khyatigandhi6** [comment]
Hi Team,
Updated the PR by resolving merge conflicts. Could you please review it?

File: .github/ALLOWLIST (hunk 1 above)
Comment (khyatigandhi6) [reply .github/ALLOWLIST]: Hi @mx-psi ,
After I removed the entry from allowlist, the check 'check-codeowners' is failing stating
> there are 1 components not included in CODEOWNERS and not known in the ALLOWLIST

Can you please guide me what should be done as no code owner has been assigned to PR yet.

Comment (mx-psi) [reply .github/ALLOWLIST]: @khyatigandhi6 Hey, as documented [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components), you need to
> Add the sponsor for your component and yourself to a new line for your component in the [.github/CODEOWNERS](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS) file.

If you are contributing a new component, you are expected to own it and respond when pinged on issues or to review PRs related to this component.

Comment (khyatigandhi6) [reply .github/ALLOWLIST]: Thanks! Made changes accordingly.

**Review by mx-psi — DISMISSED** [review]

**Comment by mx-psi** [comment]
Removing the 'request changes' since it was addressed. @bogdandrutu this is on you as the sponsor :)

**Comment by lalit-fgr** [comment]
@bogdandrutu can we please get some 👀 to it.

File: cmd/configschema/go.mod [hunk 2]
```diff
@@ -14,6 +14,8 @@ require (
 	golang.org/x/text v0.4.0
 )
 
+require github.com/logicmonitor/lm-data-sdk-go v0.5.0 // indirect
+
 require (
```
Comment (bogdandrutu) [inline cmd/configschema/go.mod:19]: Same block please

File: exporter/logicmonitorexporter/Makefile [hunk 3]
```diff
@@ -0,0 +1 @@
+include ../../Makefile.Common
```
Comment (bogdandrutu) [inline exporter/logicmonitorexporter/Makefile:1]: New line.

File: exporter/logicmonitorexporter/authenticator.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// LMAuthenticator is used for authenticating requests to Logicmonitor platform
+type LMAuthenticator struct {
+	Config *Config
+}
```
Comment (bogdandrutu) [inline exporter/logicmonitorexporter/authenticator.go:32]: Why public? Who is using this?

File: exporter/logicmonitorexporter/config.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+	exporterhelper.QueueSettings `mapstructure:"sending_queue"`
+	exporterhelper.RetrySettings `mapstructure:"retry_on_failure"`
+	ResourceToTelemetrySettings  resourcetotelemetry.Settings `mapstructure:"resource_to_telemetry_conversion"`
+
+	// ApiToken of LM Platform
+	APIToken map[string]string `mapstructure:"apitoken"`
+
+	// URL on which the logs/traces should be forwarded
+	URL string `mapstructure:"url"`
```
Comment (bogdandrutu) [inline exporter/logicmonitorexporter/config.go:43]: confighttp.HTTPClientSettings already has notion of "endpoint" this is confusing.

File: exporter/logicmonitorexporter/config.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+	// Headers for HTTP requests
+	Headers map[string]string `mapstructure:"headers"`
+
```
Comment (bogdandrutu) [inline exporter/logicmonitorexporter/config.go:47]: Already defined in HTTPClientSettings https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/confighttp.go#L57

**Review by bogdandrutu — COMMENTED** [review]
This needs more work and consider to split in simpler PRs to review.

**Comment by bogdandrutu** [comment]
Consider to follow the rules in:https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#when-adding-a-new-component

**Comment by khyatigandhi6** [comment]
Thanks @bogdandrutu for the review comments!
I will break down this PR into 3 parts.

**Pull request closed without merging**